src/server.rs

```diff
@@ -89,6 +91,21 @@ impl<A> Server<A> {
         }
     }
 
+    #[cfg(feature = "proxy-protocol")]
+    /// Enable proxy protocol header parsing.
+    /// Note has to be called after initial acceptor is set.
+    pub fn proxy_protocol_enabled(self) -> Server<ProxyProtocolAcceptor<A>> {
```

Comment:
Is this matching the builder pattern naming? Should it be `enable_proxy_protocol`?

Reply:
This call enables proxy protocol support, it doesn't check whether proxy protocol support is enabled, as the function name would imply here.

Reply:
i think `parse_proxy_header()` also works

Reply:
yes you're right now that it doesn't silently handle cases when it is not present when we switched from peeking to reading